exo/orchestration/standard_node.py

```diff
@@ -190,15 +180,108 @@ async def _process_prompt(self, base_shard: Shard, prompt: str, request_id: Opti
     shard = self.get_current_shard(base_shard)
 
     if DEBUG >= 2: print(f"[{request_id}] process prompt: {base_shard=} {shard=} {prompt=}")
-    if shard.start_layer != 0:
+    if not shard.is_first_layer():
       if DEBUG >= 2: print(f"[{request_id}] forwarding to next shard: {base_shard=} {shard=} {prompt=}")
-      await self.forward_to_next_shard(shard, prompt, request_id)
+      resp = await self.forward_prompt(shard, prompt, request_id, 0)
       return None
     else:
       result = await self.inference_engine.infer_prompt(request_id, shard, prompt)
-      ret = await self.process_result(shard, result, request_id) 
+      ret = await self.process_inference_result(shard, result, request_id) 
       return result
 
+  async def enqueue_example(
+    self,
+    base_shard: Shard,
+    example: np.ndarray,
+    target: np.ndarray, 
+    length: np.ndarray,
+    request_id: Optional[str] = None,
+    train: bool = False,
+  ):
+    shard = self.get_current_shard(base_shard)
+    if shard.is_first_layer():
+      resp = await self.process_example(shard, example, target, length, request_id)
+    else:
+      if request_id is None:
```

Comment:
That happens further down in the various process_ functions and so is unnecessary here. Basically this whole enqueue_ thing exists because unlike prompts, working with tensors doesn't have an implicit way of forwarding to the first layer, but we still want the originator of a request to be able to track its id